### ISSUE TITLE
Blender/GLTF vertex coloring is changed on import.

### ISSUE THREAD

**Issue by bertodelrio256** [godotengine/godot#92404]
### Tested versions

reproducible in v4.3.dev6.mono.official [89850d553]

### System information

Godot v4.3.dev6.mono - Windows 10.0.19044 - Vulkan (Forward+) - dedicated NVIDIA GeForce RTX 2060 (NVIDIA; 31.0.15.5161) - AMD Ryzen 9 5950X 16-Core Processor (32 Threads)

### Issue description

issue possible related to https://github.com/godotengine/godot/issues/87486. 

when exporting a mesh from blender 4.0 to using gltf, the vertex colors are not imported correctly and appear different in godot. attach is a screenshot of the vertex colors in blender and the imported result in godot. the data has been changed as you see jagged edges / harder transitions in the colors as well as slighter darker colors. (is vertex color being compressed?). if you export from blender to gltf and then reimport, you will see there is no change in the colors. (blend file attached)

![vc issue](https://github.com/godotengine/godot/assets/98141558/cf2afb58-57d4-48d1-b8a1-3ca205acd9ba)

### Steps to reproduce

-create a mesh in blender 4.0 with a range of vertex colors (colors near black seem to be problematic the most)
-export the mesh using the gltf exporter in blender
-import the glb file in godot and apply an unshaded material which displays the vertex color to albedo

### Minimal reproduction project (MRP)

[vc issue MRP.zip](https://github.com/godotengine/godot/files/15449447/vc.issue.MRP.zip)

**Comment by tiloc** [comment]
Is this potentially related to what this asset is trying to fix? https://godotengine.org/asset-library/asset/2905 changed behaviour in Blender 4.1.

**Comment by bertodelrio256** [comment]
> Is this potentially related to what this asset is trying to fix? https://godotengine.org/asset-library/asset/2905 changed behaviour in Blender 4.1.

No, thats an issue with the gltf exporter not exporting vertex color at all in 4.1. oops sorry this actually applies to blender 4.0 i meant. I will update the description.

**Comment by diegopau** [comment]
I was about to open a new issue with a problem importing a GLB model with color attributes. I have almost zero experience with Blender but I think vertex coloring is the same as using color attributes in Blender (with domain set to Vertex).
Since this is a voxel based model with a clearly defined texture (the eyes in the face and other details) it is easy to see that the colors are incorrectly applied. But if you open the advanced importer dialog in Godot then it looks fine so Godot is reading the format without issues.
![image](https://github.com/user-attachments/assets/cca23bd5-44e5-41ed-86df-ab148b43600a)
![image](https://github.com/user-attachments/assets/d97ef78e-2ce1-4102-9068-60f13707ef3b)

**Comment by Calinou** [comment]
Try disabling **Meshes > Generate LODs** in the import options.

**Comment by diegopau** [comment]
@Calinou That worked for my case, thanks!! so then my issue was a different one than the described in the first post.

**Comment by bertodelrio256** [comment]
![Untitled-1](https://github.com/user-attachments/assets/a6a73803-ddda-434a-b6ef-ce024d804293)
I just wanted to show this comparison which shows that there is way more saturated red in the imported colors in godot as seen here. This is was just done in godot 4.3 where the issue remains.

**Comment by bertodelrio256** [comment]
just a thought but, is it possible the vertex colors are being compressed somehow by godot on import?

**Comment by Calinou** [comment]
> just a thought but, is it possible the vertex colors are being compressed somehow by godot on import?

You can force disabling mesh compression in the Import options (**Force Disable Compression**).

**Comment by bertodelrio256** [comment]
> > just a thought but, is it possible the vertex colors are being compressed somehow by godot on import?
> 
> You can force disabling mesh compression in the Import options (**Force Disable Compression**).

oh? I thought that mesh compression only compressed the vertex positions. it does colors and uvs as well?

**Comment by bertodelrio256** [comment]
this is resolved. the problem was that I was passing COLOR.rgb into a varying in vertex that using that in fragment (causing it to change for some reason). when using COLOR.rgb directly in fragment that issue is gone.
